java/src/main/java/api/example/java/Config.java

```diff
@@ -0,0 +1,88 @@
+package api.example.java;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.util.Base64Utils;
+import org.springframework.util.StringUtils;
+import org.springframework.web.util.UriComponentsBuilder;
+
+@Configuration
+public class Config {
+    @Value("${climate.login.server}")
+    public String loginServer;
+
+    @Value("${climate.login.path}")
+    public String loginPath;
+
+    @Value("${climate.token.server}")
+    public String tokenServer;
+
+    @Value("${climate.token.path}")
+    public String tokenPath;
+
+    @Value("${climate.api.server}")
+    public String apiServer;
+
+    @Value("${CLIENT_ID}")
+    public String clientId;
+
+    @Value("${CLIENT_SECRET}")
+    public String clientSecret;
+
+    @Value("${API_KEY}")
+    public String apiKey;
+
+    @Value("${API_SCOPES}")
+    public String scopes;
+
+    public Config() {
+        if (StringUtils.isEmpty(System.getenv("CLIENT_ID"))) {
+            throw new IllegalArgumentException("Please set environment variable CLIENT_ID");
+        }
+        if (StringUtils.isEmpty(System.getenv("CLIENT_SECRET"))) {
+            throw new IllegalArgumentException("Please set environment variable CLIENT_SECRET");
+        }
+        if (StringUtils.isEmpty(System.getenv("API_KEY"))) {
+            throw new IllegalArgumentException("Please set environment variable API_KEY");
+        }
+        if (StringUtils.isEmpty(System.getenv("API_SCOPES"))) {
+            throw new IllegalArgumentException("Please set environment variable API_SCOPES");
+        }
+    }
+
+    public String buildOauthLink(String server) {
+        /*
+         * https://climate.com/static/app-login/index.html?scope=${scope}
+         * &page=oidcauthn&response_type=code&redirect_uri=${redirect_uri}
+         * &client_id=${client_id}
+         */
+        return UriComponentsBuilder.newInstance().scheme("https").host(loginServer).path(loginPath)
```

Comment:
IMO it'd be more readable to have each method call on it's own line, like:
```
UriComponentsBuilder.newInstance()
    .scheme("https")
    .host(loginServer)
    .path(loginPath)
    ....
```

Reply:
It is eclipse default behavior. fixed now.